File: algorithm/杭电ACM/ambition0109/1060.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-07-03 16:38:40
////Compiler: GUN C++
//////////////////////////////////////////////////////
////Problem ID: 1060
////Problem Title: 
////Run result: Accept
////Run time:0MS
////Run memory:372KB
//////////////////System Comment End//////////////////
/*
 * 1060.cpp
 *
 *  Created on: 2010-7-3
 *      Author: ambition
 */

#include<iostream>
#include<cmath>
using namespace std;

int main()
{
	int n,m;
	double g;
	double h;
	for(cin>>n;n>0;n--)
	{
		cin>>m;
		g=m*log10((double)m);
		g=g-(long long)g;
		h=pow(10.,(double)g);
		cout<<(int)h<<endl;
	}
	return 0;
}

File: algorithm/HDU-ACM/ambition0109/1060.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 21:29:11
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1060
////Problem Title: 
////Run result: Accept
////Run time:109MS
////Run memory:6196KB
//////////////////System Comment End//////////////////
import java.util.Scanner;

public class Main {
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		int t=in.nextInt();
		while(t--!=0){
			int n=in.nextInt();
			double g=n*Math.log10((double)n);
			g=g-(long)g;
			double h=Math.pow(10.0,g);
			System.out.println(((int)h));
		}
	}
}
